apps/emqx_schema_registry/src/emqx_schema_registry_serde.erl

```diff
@@ -108,35 +147,79 @@ make_serde(protobuf, Name, Source) ->
         name = Name,
         type = protobuf,
         eval_context = SerdeMod
-    }.
+    };
+make_serde(json, Name, Source) ->
+    case json_decode(Source) of
+        SchemaObj when is_map(SchemaObj) ->
+            %% jesse:add_schema adds any map() without further validation
+            %% if it's not a map, then case_clause
+            ok = jesse_add_schema(Name, SchemaObj),
+            #serde{name = Name, type = json};
+        _NotMap ->
+            error({invalid_json_schema, bad_schema_object})
+    end.
 
 eval_decode(#serde{type = avro, name = Name, eval_context = Store}, [Data]) ->
     Opts = avro:make_decoder_options([{map_type, map}, {record_type, map}]),
     avro_binary_decoder:decode(Data, Name, Store, Opts);
 eval_decode(#serde{type = protobuf, eval_context = SerdeMod}, [EncodedData, MessageName0]) ->
     MessageName = binary_to_existing_atom(MessageName0, utf8),
     Decoded = apply(SerdeMod, decode_msg, [EncodedData, MessageName]),
-    emqx_utils_maps:binary_key_map(Decoded).
+    emqx_utils_maps:binary_key_map(Decoded);
+eval_decode(#serde{type = json, name = Name}, [Data]) ->
+    true = is_binary(Data),
+    Term = json_decode(Data),
+    {ok, NewTerm} = jesse_validate(Name, Term),
+    NewTerm.
 
 eval_encode(#serde{type = avro, name = Name, eval_context = Store}, [Data]) ->
     avro_binary_encoder:encode(Store, Name, Data);
 eval_encode(#serde{type = protobuf, eval_context = SerdeMod}, [DecodedData0, MessageName0]) ->
     DecodedData = emqx_utils_maps:safe_atom_key_map(DecodedData0),
     MessageName = binary_to_existing_atom(MessageName0, utf8),
-    apply(SerdeMod, encode_msg, [DecodedData, MessageName]).
+    apply(SerdeMod, encode_msg, [DecodedData, MessageName]);
+eval_encode(#serde{type = json, name = Name}, [Map]) ->
+    %% The input Map may not be a valid JSON term for jesse
+    Data = iolist_to_binary(emqx_utils_json:encode(Map)),
+    NewMap = json_decode(Data),
+    case jesse_validate(Name, NewMap) of
+        {ok, _} ->
+            Data;
```

Comment:
Minor nit:

```suggestion
        {ok, NewData} ->
            NewData;
```

Reply:
we don't pass a `parser_fun`, in `Options` for `jesse:validate/3`.
so it should be the exact same value returned.